ssz-rs/src/bitlist.rs

```diff
@@ -159,14 +161,22 @@ impl<const N: usize> Deserialize for Bitlist<N> {
         }
 
         let (last_byte, prefix) = encoding.split_last().unwrap();
+        if *last_byte == 0u8 {
+            return Err(DeserializeError::InvalidByte(*last_byte));
+        }
+
         let mut result = BitlistInner::from_slice(prefix);
         let last = BitlistInner::from_element(*last_byte);
+        // checked subtraction is unnecessary, as last_byte != 0, so last.trailing_zeros <= 7; qed
```

Comment:
another thing we could do here is to make it clear these comments are for _safety_

```suggestion
        // SAFETY: checked subtraction is unnecessary, as last_byte != 0, so last.trailing_zeros <= 7; qed
```